### PR TITLE
[Upgrade] Various fixes

### PR DESCRIPTION
**Pull request by bk201**
This PR improves the upgrade node script by:

- [Upgrade: shutdown VMs if live-migration takes too long](https://github.com/harvester/harvester/commit/c26f323205b12b0363706663654a9af7bc284984)
  - Could be a workaround to https://github.com/harvester/harvester/issues/2029.
- [Upgrade: image enhancement](https://github.com/harvester/harvester/commit/db7a56406056b71a42cbc7ab19b74c53cbe480de)
  - Clean RKE2 image tarball when upgrade is done.
  - Preload RKE2 images. RKE2 images might be needed during nodes upgrade. A waiting-for-upgrade node doesn't load RKE2 images yet because its RKE2 server/agent doesn't restart. We need to preload RKE2 images beforehand.

**Related Issue:**
https://github.com/harvester/harvester/issues/2029
https://github.com/harvester/harvester/issues/1861

**Test plan:**
- Perform an upgrade
- There should be no RKE2 tarballs under the images directory.
  ```
  node1:~ # ls /var/lib/rancher/rke2/agent/images/rke2*
  ls: cannot access '/var/lib/rancher/rke2/agent/images/rke2*': No such file or directory
  ```

### CODE REVIEW

File: package/upgrade/upgrade_node.sh [hunk 1]
```diff
@@ -116,6 +117,30 @@ wait_vms_out()
   done
 }
 
+wait_vms_out_or_shutdown()
+{
+  local vm_count
+  local max_retries=240
```
Comment (weihanglo) [inline package/upgrade/upgrade_node.sh:100]: Just wonder why 240 times of retry?

**Review by weihanglo — APPROVED** [review]
LGTM

File: package/upgrade/upgrade_node.sh (hunk 1 above)
Comment (bk201) [reply package/upgrade/upgrade_node.sh:100]: it's around 20 minutes :) (240 *5)

**Review by FrankYang0529 — APPROVED** [review]
LGTM.

**Review by weihanglo — APPROVED** [review]
LGTM. Those images were deleted successfully.

**Pull request merged**